insights/protocol.py

```diff
@@ -131,12 +135,16 @@ def deserialize(self) -> Dict:
         return self.output
 
 class Challenge(BaseSynapse):
+    model_type: str # model type
```

Comment:
maybe we should add here challenge type? then we can reuse MODEL_TYPE_BALANCE_TRACKING and MODEL_TYPE_FUNDS_FLOW